docs/source/python/integration/python_r.rst

```diff
@@ -0,0 +1,315 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+Integrating PyArrow with R
+==========================
+
+Arrow supports exchanging data within the same process through the
+:ref:`c-data-interface`.
+
+This can be used to exchange data between Python and R functions and
+methods so that the two languages can interact without any cost of
+marshaling and unmarshaling data.
+
+.. note::
+
+    The article takes for granted that you have a ``Python`` environment
+    with ``pyarrow`` correctly installed and an ``R`` environment with
+    ``arrow`` library correctly installed.
```

Comment:
Could we? Should we? link to https://arrow.apache.org/docs/python/install.html and https://arrow.apache.org/docs/r/#installation here?

Reply:
Added link, I guess we could also remove this note. I felt it was useful mostly because if you approach it from a Python developer point of view it might not be immediately obvious that you need a working R environment too. But in the end we don't state on every doc page that to follow the doc you need a working Arrow setup.

Reply:
Thanks for this! I'm happy either way (to leave it in or take it out). It _is_ nice to have the links for building both in case one or the other is less familiar to someone one – but you're absolutely right we don't start each doc with "to build Arrow..."